### PR TITLE
[doc] Update dev_install.md to reflect build.py usage

### PR DESCRIPTION
**Pull request by feisuzhu**
Issue: #

### <samp>🤖 Generated by Copilot at b6b396e</samp>

This pull request improves the documentation for developers who want to build and install Taichi from source. It explains how to use the `build.py` script and its options, and simplifies the steps for different platforms and backends.

### CODE REVIEW

File: docs/lang/articles/contribution/dev_install.md [hunk 1]
```diff
@@ -32,117 +35,82 @@ Installation instructions vary depending on which operating system (OS) you are
...
-| Python                       | 3.7/3.8/3.9/3.10 We recommend installing Python from [Miniforge](https://github.com/conda-forge/miniforge/#download) conda if you are on a MacBook with M1 chip. |
-| Clang++                      | Clang++ &gt;8                            |
-| LLVM                         | 15.0.4 (Taichi customized version)                                                                                                                                                       |
-| Command line tools for Xcode | For macOS users only: `xcode-select --install `                                                                                                                                          |
+| Category                      | Prerequisites                                                        |
+| :---------------------------- | :------------------------------------------------------------------- |
+| Linux distribution            | Anything recent enough, e.g. Ubuntu 20.04                            |
+| Python                        | 3.6+, with a usable distutils(`python3-distutils` package on Ubuntu) |
```
Comment (jim19930609) [inline docs/lang/articles/contribution/dev_install.md:63]: Do we support >python3.10?

File: docs/lang/articles/contribution/dev_install.md [hunk 2]
```diff
@@ -32,117 +35,82 @@ Installation instructions vary depending on which operating system (OS) you are
...
 
-<TabItem value="linux">
+Taichi supports building from source with Clang++ >= 10.0 and MSVC from VS2022.
 
-1. Set `LLVM_DIR` environment variable:
-   ```
-   echo "export LLVM_DIR=<PATH_TO_LLVM_FOLDER>" >>  ~/.bashrc
-   ```
-2. Update your path for the remainder of the session:
+For macOS developers, it is recommended to use AppleClang, which comes with the Command Line Tools for Xcode. You can install them by running `xcode-select --install`. Alternatively, a complete installation of Xcode.app is also an option.
```
Comment (jim19930609) [inline docs/lang/articles/contribution/dev_install.md:98]:
```suggestion
For macOS developers, it is recommended to use AppleClang, which comes with the Command Line Tools for Xcode. You can install them by running `xcode-select --install`. Alternatively, you can also install Xcode.app from the Apple Store.
```

File: docs/lang/articles/contribution/dev_install.md [hunk 3]
```diff
@@ -341,22 +321,56 @@ You must install the Vulkan SDK in order to debug Taichi's Vulkan backend. To pr
...
+
+  # export DEBUG=1 # Uncomment it if you wish to keep debug information.
+
+  # This would build a wheel file in `dist/` folder
+  ./build.py
+
+  # Alternatively, this would drop into a shell with complete build environment,
+  ./build.py --shell
+  # and then you could install Taichi in development mode
+  python3 setup.py develop
```
Comment (jim19930609) [inline docs/lang/articles/contribution/dev_install.md:344]: I'd suggest put the alternative method in a separate section/code-block, so it's not gonna confuse the beginners.

**Review by jim19930609 — APPROVED** [review]

File: docs/lang/articles/contribution/dev_install.md (hunk 1 above)
Comment (feisuzhu) [reply docs/lang/articles/contribution/dev_install.md:63]: Yes, `3.6+` means 3.6 or later

**Pull request merged**